Userland/Libraries/LibWeb/DOM/Node.cpp

```diff
@@ -1587,4 +1590,235 @@ void Node::build_accessibility_tree(AccessibilityTreeNode& parent) const
+    return name_or_description(NameOrDescription::Name, document, visited_nodes);
+}
+
+// https://www.w3.org/TR/accname-1.2/#mapping_additional_nd_description
+ErrorOr<String> Node::accessible_description(Document const& document) const
+{
+    // If aria-describedby is present, user agents MUST compute the accessible description by concatenating the text alternatives for elements referenced by an aria-describedby attribute on the current element.
+    // The text alternatives for the referenced elements are computed using a number of methods, outlined below in the section titled Accessible Name and Description Computation.
+    if (is_element()) {
+        HashTable<i32> visited_nodes;
+        StringBuilder builder;
+        auto const* element = static_cast<Element const*>(this);
+        auto id_list = element->aria_described_by().split_view(Infra::is_ascii_whitespace);
+        for (auto const& id : id_list) {
+            if (auto description_element = document.get_element_by_id(id)) {
+                auto description = TRY(description_element->name_or_description(NameOrDescription::Description, document, visited_nodes));
+                if (!description.is_empty()) {
+                    if (builder.is_empty()) {
+                        builder.append(description);
+                    } else {
+                        builder.append(" "sv);
+                        builder.append(description);
+                    }
+                }
+            }
+        }
+        return builder.to_string();
+    }
+    return String {};
+}
+
+Optional<StringView> Node::first_valid_id(DeprecatedString const& value, Document const& document)
+{
+    auto id_list = value.split_view(Infra::is_ascii_whitespace);
+    for (auto const& id : id_list) {
+        if (document.get_element_by_id(id))
+            return id;
+    }
+    return {};
+}
+
+// https://www.w3.org/TR/accname-1.2/#mapping_additional_nd_te
+void Node::append_without_space(StringBuilder X, StringView const& result)
+{
+    // If X is empty, copy the result to X.
+    // If X is non-empty, copy the result to the end of X.
+    X.append(result);
+}
+
+// https://www.w3.org/TR/accname-1.2/#mapping_additional_nd_te
+void Node::append_with_space(StringBuilder X, StringView const& result)
+{
+    // If X is empty, copy the result to X.
+    if (X.is_empty()) {
+        X.append(result);
+    }
+    // If X is non-empty, add a space to the end of X and then copy the result to X after the space.
+    else {
+        X.append(" "sv);
+        X.append(result);
+    }
+}
+
+// https://www.w3.org/TR/accname-1.2/#mapping_additional_nd_te
+void Node::prepend_without_space(StringBuilder X, StringView const& result)
+{
+    // If X is empty, copy the result to X.
+    if (X.is_empty()) {
+        X.append(result);
+    }
+    // If X is non-empty, copy the result to the start of X.
+    else {
```

Comment:
This formatting is a bit odd with a comment between two branches of an `if`. Maybe move this into the else branch (same for the one below & above)